judo-ui-react/src/main/resources/actor/src/components/table/LazyTable.tsx.hbs

```diff
@@ -0,0 +1,814 @@
+      setLastItem(res[res.length - 1]);
+      setRowCount(res.length || 0);
+      } catch (error) {
+        handleError(error);
+      } finally {
+        setIsInternalLoading(false);
+      }
+
+  }
+
+  async function fetchData() {
+    if (!containerHasTable) {
+      if (ownerData && ownerData.__signedIdentifier) {
+        await fetching();
+      }
+    } else {
+      await fetching();
+    }
+  }
+
+  useEffect(() => {
+    fetchData();
+    handleOnSelection(selectionModel);
+  }, [queryCustomizer, refreshCounter]);
+
+  let additionalToolbarActions: Record<string, ElementType>;
+
+  if (actions.AdditionalToolbarButtons) {
+    if (containerHasTable) {
+      additionalToolbarActions = actions.AdditionalToolbarButtons(
+        data,
+        isLoading,
+        getSelectedRows(),
+        clearSelections,
+      )
+    } else {
+      additionalToolbarActions = actions.AdditionalToolbarButtons(
+        data,
+        isLoading,
+        getSelectedRows(),
+        clearSelections,
+        ownerData,
+        editMode,
+        isFormUpdateable,
+      )
+    }
+  } else {
+    additionalToolbarActions = {};
+  }
+
+  const AdditionalToolbarActions = () => {
+    return <>
+      { Object.keys(additionalToolbarActions).map((key) => {
+        const AdditionalButton = additionalToolbarActions[key];
+        return <AdditionalButton key={key} />;
+      }) }
+    </>;
+  };
+
+  async function filterAction(id: string) {
+    const filterResults = await actions[calculateActionName(relationName, 'filterAction')]!(id, filterOptions, filterModel, filters);
+  {{# if isUseInlineColumnFilters }}
+    if (filterResults.model) {
+      handleFilterModelChange({
+        ...filterResults.model!,
+      });
+    }
+  {{ else }}
+    if (Array.isArray(filterResults.filters)) {
+      handleFiltersChange([...filterResults.filters!]);
+    }
+  {{/ if }}
+  }
+
+  const getTableButtonVisibility = useCallback((expression: string, actionName: string): boolean => {
+    if (expression === undefined) {
+      return false;
+    }
+
+    if (expression === 'false') {
+      return false;
+    }
+
+    if (expression === 'isFormUpdateable()') {
+      return isFormUpdateable ? isFormUpdateable() : false;
+    }
+
+    let result: boolean = data.length !== undefined && data.length > 0;
+    if (expression.includes(`ownerData.${enabledByName}`)) {
+      result = result && ownerData && ownerData['enabledByName'];
+    }
+
+    if (expression.includes('isFormUpdateable()')) {
+      return result && (isFormUpdateable ? isFormUpdateable(): false);
+    }
+
+    if (actionName.includes('clearAction') || actionName.includes('ClearAction')) {
+      return result;
+    }
+
+    if (expression === 'selectionModel.length > 0') {
+      return selectionModel.length > 0;
+    }
+
+    return true;
+  }, [selectionModel, ownerData, isFormUpdateable, data]);
+
+  const getTableRowButtonDisabledConditions = useCallback((row: T): boolean => {
```

Comment:
We need to come up with a version which is more similar to the callback approach.

Reply:
[f51c46b](https://github.com/BlackBeltTechnology/judo-ui-react-template/pull/365/commits/f51c46bc6ce006871f020c8d52167c195794f251)